src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ShellViewModel.cs

```diff
@@ -20,39 +16,28 @@ public partial class ShellViewModel(IServiceProvider _serviceProvider) : Observa
     [ObservableProperty]
     public partial bool IsLoaded { get; set; } = false;
 
-    public ObservableCollection<CommandProviderWrapper> ActionsProvider { get; set; } = [];
-
-    public ObservableCollection<ExtensionObject<IListItem>> TopLevelCommands { get; set; } = [];
-
-    private IEnumerable<ICommandProvider>? _builtInCommands;
-
     [RelayCommand]
     public async Task<bool> LoadAsync()
     {
-        _builtInCommands = _serviceProvider.GetServices<ICommandProvider>();
-
-        // Load Built In Commands First
-        foreach (var provider in _builtInCommands)
-        {
-            CommandProviderWrapper wrapper = new(provider);
-            ActionsProvider.Add(wrapper);
-
-            await LoadTopLevelCommandsFromProvider(wrapper);
-        }
-
+        var tlcManager = _serviceProvider.GetService<TopLevelCommandManager>();
+        await tlcManager!.LoadBuiltinsAsync();
         IsLoaded = true;
 
-        // TODO: would want to hydrate this from our services provider in the View layer, need to think about construction here...
-        WeakReferenceMessenger.Default.Send<NavigateToListMessage>(new(new(new MainListPage(this))));
-        return true;
-    }
+        // Built-ins have loaded. We can display our page at this point.
+        var page = _serviceProvider.GetService<MainListPage>();
+        WeakReferenceMessenger.Default.Send<NavigateToListMessage>(new(new(page!)));
```

Comment:
Yeah our 'MainListPage' object is a bit weird, but I don't think we want to grab it from the service index. This is probably a remnant of how we just started piecing this together during the hackathon, so we may want to evaluate how we want navigation to work and where our initial starting 'model' comes from to kick us off.